src/main/objects/Parser.js

```diff
@@ -19,12 +19,14 @@ function parseChord(string) {
     var chord_type = string.substring(0, i)
     var arguments = string.substring(i, string.length)
 
-    if (arguments === null || arguments.length < 2) {
+    if (arguments === null || arguments.length < 2 || chord_type.length > 2) {
         return undefined
     }
-
-    var arguments_json = JSON.parse(arguments)
-    arguments_json["functionName"] = chord_type;
+    var mode = Consts.MODE.MAJOR;
+    if(chord_type.length === 2 && chord_type[1] === "o") mode = Consts.MODE.MINOR;
+    var arguments_json = JSON.parse(arguments);
+    arguments_json["functionName"] = chord_type[0];
+    arguments_json["mode"] = mode;
```

Comment:
Wyniosłeś odpowiedzianość wartości defaultowych do parsera. Lepiej będzie jeśli będzie minor w przypadku kółka w nazwie akordu, a undefined wpp. Może nie znajduje to bezpośredniego zastosowania jednak wydaje mi się że lepiej żeby ta odpowiedzianość nie była rozproszona. Można też przedyskutować ;)

Reply:
Poprawione